pyroute2/netlink/rtnl/tcmsg/act_connmark.py

```diff
@@ -0,0 +1,45 @@
+from pyroute2.netlink import nla
+from pyroute2.netlink import NLA_F_NESTED
+from pyroute2.netlink.rtnl.tcmsg.common import tc_actions
+
+"""
+connmark - netfilter connmark retriever action
+see tc-connmark(8)
+
+This filter restores the connection mark into the packet mark.
+Connection marks are typically handled by the CONNMARK iptables module.
+See iptables-extensions(8).
+
+There is no mandatory parameter, but you can specify the action, which defaults
+to 'pipe', and the conntrack zone (see the manual).
+"""
+
+
+class options(nla):
+    nla_flags = NLA_F_NESTED
+    nla_map = (('TCA_CONNMARK_UNSPEC', 'none'),
+               ('TCA_CONNMARK_PARMS', 'tca_connmark_parms'),
+               ('TCA_CONNMARK_TM', 'none'),
+               )
+
+    class tca_connmark_parms(nla):
+        fields = (('index', 'I'),
+                  ('capab', 'I'),
+                  ('action', 'i'),
+                  ('refcnt', 'i'),
+                  ('bindcnt', 'i'),
+                  ('zone', 'H'),
+                  ('__padding', 'H'),  # XXX is there a better way to do this ?
```

Comment:
Not yet. And unfortunately I'm not sure there may be a better way — the structure fields alignment sometimes doesn't work as expected neither in `struct`, nor in `ctypes`. So I would leave this as is, with `__padding`. May be stupid, but portable.